In [ ]:
import random
import pickle
import numpy as np
import pandas as pd
from datetime import datetime
#from sklearn.preprocessing import OneHotEncoder

## Here take mtags file where member id AND IT'S TAGS ARE PRESENT

In [ ]:
ct=0
mtags=pickle.load(open('mtags','r'))
print(len(mtags))
for m in mtags:
    print(m,mtags[m])
    if ct>5:
        break
    ct+=1

423763
(57671682, set([u'Dining Out', u"Singles 30's-50's", u'Dating and Relationships', u"Women's Health and Wellness", u"Women's Social", u"Women's Self-Defense"]))
(100663302, set([]))
(209715208, set([u'Collaboration between Creative Minds', u'Bicycling', u'Travel', u'Cooking', u'Creativity', u'Parents', u'Self-Improvement', u'Pets', u'Single Parent Support', u'Dogs', u'Live Music', u'Shopping', u'Moms', u'Healthy Living', u'Crafts', u'Work at Home Moms', u'Divorced Parents', u'Fun Times', u'Paper Crafts', u'Dining Out', u'Dating and Relationships', u'Small Breed Dogs', u'Steampunk', u'Eating, Drinking, Talking, Laughing, Etc', u'Fashion and Style', u'Musicians', u'Historic Buildings', u'Outdoors', u'Hiking', u'Conversation', u'Playdates', u'Fun and Laughter', u'Safe Skin Care & Cosmetics', u'Whole Food Nutrition', u'Artists', u'Music', u'Meditation', u'Dog Playgroups']))
(127926282, set([u'SEO (Search Engine Optimization)', u'WordPress', u'WordPress Small Business Web Sites']))
(1

## Here mem_at file take where member id and their attend events and duration given

In [ ]:

#It gives the set of (event-id, event_time) for all the events attended by each user.
ct=0
mem_at=pickle.load(open('mem_at','r'))
print(len(mem_at))
for m in mem_at:
    print("NEW ID  ",m,mem_at[m])
    if ct>5:
        break
    ct+=1

223788
('NEW ID  ', 57671682, set([(u'74913292', 1344724200000L)]))
('NEW ID  ', 188219397, set([(u'227073685', 1450224000000L)]))
('NEW ID  ', 100663302, set([(u'dhzjmgyrnbnc', 1383175800000L), (u'dhzjmgyrlblc', 1377732600000L), (u'dhzjmgyrnbdb', 1380756600000L), (u'dhzjmgyrqbxb', 1387413000000L), (u'dhzjmgyrlbcc', 1377127800000L), (u'dhzjmgyrnbvb', 1381966200000L), (u'dhzjmgyrlbsb', 1376523000000L)]))
('NEW ID  ', 90701832, set([(u'166281682', 1400626800000L)]))
('NEW ID  ', 200802315, set([(u'ghsfrlyvfbnc', 1459381500000L)]))
('NEW ID  ', 152043532, set([(u'xvgznyskbdb', 1404345600000L)]))
('NEW ID  ', 180180312, set([(u'211533242', 1414884600000L)]))


In [ ]:
##checking mem_at keys are same as mtags keys
ct=0
keys = list(mtags.keys())
#ename_CH=pickle.load(open('ename_bhnoda_155_CH','r'))
#print(len(ename_CH))
for m in mem_at:
    print(m)
    if m in keys:
        print("True")
    else: print("false")
    if ct>5:
        break
    ct+=1

57671682
True
188219397
True
100663302
True
90701832
True
200802315
True
152043532
True
180180312
True


In [ ]:
##Here I taking the keys of mtags and make a list and set
s=[]
for m in mem_at:
    #print(m,mtags[m])
    s.append(m)
print(len(s))
print(s[0:5])
print(len(set(s)))
unique_s=set(s)

223788
[57671682, 188219397, 100663302, 90701832, 200802315]
223788


In [ ]:
## Here I taking values of mem_at file.
ct=0
l=[]
for m in mem_at:
    #print(m,mtags[m])
    l.append(list(mem_at[m]))
print(len(l))
#print((l[5]))
for i in range(5):
    print((l[i]))
    print(len(l[i]))

223788
[(u'74913292', 1344724200000L)]
1
[(u'227073685', 1450224000000L)]
1
[(u'dhzjmgyrnbnc', 1383175800000L), (u'dhzjmgyrlblc', 1377732600000L), (u'dhzjmgyrnbdb', 1380756600000L), (u'dhzjmgyrqbxb', 1387413000000L), (u'dhzjmgyrlbcc', 1377127800000L), (u'dhzjmgyrnbvb', 1381966200000L), (u'dhzjmgyrlbsb', 1376523000000L)]
7
[(u'166281682', 1400626800000L)]
1
[(u'ghsfrlyvfbnc', 1459381500000L)]
1


In [ ]:
#Here I taking all events and their duration in a list
k=[]
for i in range(len(l)):


    for j in range(len(l[i])):
        #m.append(s[i])
        k.append(l[i][j])
print("len of k= ",len(k))
#print("len of m= ",len(m))
k[0:8]
set_k=set(k)
# l1=list(set(k))
# print(len(set(k)))

('len of k= ', 2261042)


In [ ]:
len(k)

2261042

## Here I take 'edata' file where has event id and place id

In [ ]:

#It gives the set of (event-id, event_place id) for all the events attended by each user.
ct=0
edata=pickle.load(open('edata','r'))
print(len(edata))
for m in edata:
    print(m,edata[m])
    if ct>5:
        break
    ct+=1

581546
(u'dlfrldysgbbc', 4410512)
(u'222202616', None)
(u'hrkzllyvfbjc', 1308946)
(u'170374282', 250527)
(u'120420672', None)
(u'222202619', None)
(u'qvhhnyqcblc', None)


In [ ]:
##clear the none values
d={}
for i in edata:
    if edata[i]!=None:
        d[i]=edata[i]
print(len(d))
ct=0
for m in d:
    print(m,d[m])
    if ct>5:
        break
    ct+=1

463897
(u'dlfrldysgbbc', 4410512)
(u'hrkzllyvfbjc', 1308946)
(u'170374282', 250527)
(u'228592712', 24296810)
(u'njvctlyvlbhc', 839555)
(u'dpqbgcyrnbqb', 1316418)
(u'dxzpffywjbfc', 250696)


In [ ]:
##taking keys of event place file
s1=[]
for m in d:
    #print(m,mtags[m])
    s1.append(m)
print(len(s1))
print(s1[0:5])
print(len(set(s1)))
unique_s1=set(s1)

463897
[u'dlfrldysgbbc', u'hrkzllyvfbjc', u'170374282', u'228592712', u'njvctlyvlbhc']
463897


## New 'et' file of event and time& date file

In [ ]:
"""
It is a dictionary which stores the starting times of the events given its id.
It is in unix epoch format. To convert to datetime, may use something like
- https://stackoverflow.com/questions/21787496/converting-epoch-time-with
-milliseconds-to-datetime
"""
ct=0
et=pickle.load(open('et','r'))
print(len(et))
for m in et:
    print(m,et[m])
    if ct>5:
        break
    ct+=1

430406
(u'dlfrldysgbbc', 1398015000000L)
(u'222202616', 1431475200000L)
(u'170374282', 1403366400000L)
(u'120420672', 1370098800000L)
(u'222202619', 1431561600000L)
(u'qvhhnyqcblc', 1327770000000L)
(u'dpqbgcyrnbqb', 1381586400000L)


In [ ]:
##taking common element of e_time and mem_at file event attend by customer file
ct=0
k1=[]
#et=pickle.load(open('et','r'))
print(len(et))
for m in et:
    k1.append((m,et[m]))

430406


In [ ]:
k1[:5]
set_k1=set(k1)

In [ ]:
print(len(set_k1),len(set_k))

(430406, 401007)


In [ ]:
## Taking common of 'mem_at' value and "et" file data,which has (event id,duration)
common1=(list(set_k1.intersection(set_k)))
len(common1)
#common1[:5]

401007

In [ ]:
#taking event id only and left the duration
event_id=[]
for i in range(len(common1)):
    event_id.append(common1[i][0])
unique_event_id=set(event_id)

## Here is e_duration file which has duration

In [ ]:
#It is a dictionary giving the duration of each event given its id.

ct=0
e_duration=pickle.load(open('e_duration','r'))
print(len(e_duration))
for m in e_duration:
    print(m,e_duration[m])
    if ct>5:
        break
    ct+=1

156617
(u'178787772', 5400000)
(u'233043815', 28800000)
(u'225522343', 21600000)
(u'219997281', 1800000)
(u'179620872', 7200000)
(u'170374282', 21600000)
(u'wgsfllytqbwb', 7200000)


In [ ]:
#Here we make duration in hour
from __future__ import division
for m in e_duration:
    e_duration[m]=round(e_duration[m]/3600000)

In [ ]:
#making subset of et
set_keysOfDuration=set(e_duration.keys())

## New file of event description

In [ ]:
# It gives the description of the events
ct=0
edesc=pickle.load(open('edesc_bhnoda_155','r'))
print(len(edesc))
for m in edesc:
    print("NEW ID  ",m,edesc[m])
    print("\n")
    if ct>5:
        break
    ct+=1

844249
('NEW ID  ', u'222202616', u'<p><a href="https://vimeo.com/113828701" class="embedded">https://vimeo.com/113828701</a></p> <p><b>Learn how to make a professional website with Wordpress</b></p> <p><br/>~~~</p> <p>Important Note: This is an online class that you take at your own time, at your own pace and from anywhere in the World.\xa0</p> <p>Purchase Class:\xa0<a href="http://online-classes.startupsaturdays.net/product/learn-to-make-your-own-business-or-personal-website/"><a href="http://online-classes.startupsaturdays.net/product/learn-to-make-your-own-business-or-personal-website/" class="linkified">http://online-classes.startupsaturdays.net/product/learn-to-make-your-own-business-or-personal-website/</a></a></p> <p>~~~</p> <p>There are no words to explain how important this class is. If you are ever going to only take 1 class in web development then this is the class you want to take. WordPress is revolutionary, its a trend setter, its the future, its the stuff that all those

In [ ]:
##taking event id and make a list and set
keys_edesc=[]
for m in edesc:
    #print(m,mtags[m])
    keys_edesc.append(m)
print(len(keys_edesc))
print(keys_edesc[0:5])
print(len(set(keys_edesc)))
unique_keys_edesc=set(keys_edesc)

844249
[u'222202616', u'kbqgvpyxjbvb', u'jpstnmywdbqb', u'wvbbqqyzcbwb', u'120420672']
844249


## Here I take "ename_bhnoda_155_CH" which has event names

In [ ]:
ct=0
ename_CH=pickle.load(open('ename_bhnoda_155_CH','r'))
print(len(ename_CH))
for m in ename_CH:
    print(m,ename_CH[m])
    if ct>5:
        break
    ct+=1

872831
(u'222202616', u'Learn to make a professional website - Training with Certificate - Online')
(u'kbqgvpyxjbvb', u'Google Business Listing')
(u'jpstnmywdbqb', u'Bachata I (Session starts 02/05/2017)')
(u'wvbbqqyzcbwb', u'Thursday Basketball Pickup at WCF - 9:30')
(u'120420672', u'Saturday morning run!')
(u'222202619', u'Learn to make a professional website - Training with Certificate - Online')
(u'qvhhnyqcblc', u'Dance Bootcamp: Salsa & Bachata, Emphasis on Ladies Styling & Great Combos')


In [ ]:
##taking event id from "ename" file and make a list and set
keys_ename=[]
for m in ename_CH:
    #print(m,mtags[m])
    keys_ename.append(m)
print(len(keys_ename))
print(keys_ename[0:5])
print(len(set(keys_ename)))
unique_keys_ename=set(keys_ename)

872831
[u'222202616', u'kbqgvpyxjbvb', u'jpstnmywdbqb', u'wvbbqqyzcbwb', u'120420672']
872831


## Making common list of events

In [ ]:
#intersection of mem_at,e_description and e_duration and e_place and et file also
## event id is in common
common=(list(unique_event_id.intersection(unique_s1,unique_keys_edesc,unique_keys_ename,set_keysOfDuration)))
len(common)
#common[:5]

115571

In [ ]:
## Here I create a dict of common event and event place id
common_event={}
for i in common:
    common_event[i]=d[i]

In [ ]:
len(common_event)

115571

In [ ]:
#initilizing the **values** of the dict
place_key=[]
for m in common_event:
    #print(m,mtags[m])
    place_key.append(common_event[m])
print(len(place_key))
#print((l[5]))
# for i in range(5):
#     print((l[i]))
#     print(len(l[i]))
(place_key[0:5])

115571


[5961062, 23651107, 250527, 876837, 23399902]

In [ ]:
##Taking counter of most common place key
from collections import Counter
hset=Counter(place_key);
most_place_key=hset.most_common(1000)
(most_place_key[0:10])

[(5274352, 1281),
 (15776612, 1195),
 (24187484, 1029),
 (5961062, 965),
 (21056022, 861),
 (7049402, 756),
 (23868614, 649),
 (1151745, 518),
 (1544507, 473),
 (4777102, 446)]

In [ ]:
#making list from dictionary of most common place id
most_place_key_list=[]
for i in range(len(most_place_key)):
    most_place_key_list.append(most_place_key[i][0])

In [ ]:
import copy
#new_list = copy.copy(old_list)
copy_most_place_key=copy.copy(most_place_key_list)
#copy_unique_place_key

In [ ]:
##Creating event id and place id dictionary
import copy
common1_event={}
for m in common_event:
    copy_most_place_key=copy.copy(most_place_key_list)
    for i in range(len(most_place_key_list)):
        if most_place_key_list[i]==common_event[m]:
            copy_most_place_key[i]=1
        else:
            copy_most_place_key[i]=0
    common1_event[m]=copy_most_place_key
#     ct+=1
#     if ct>5:
#         break

KeyboardInterrupt: 

## Here is the location vectors

In [ ]:
ct=0
for m in common1_event:
    print(m,common1_event[m])
    if ct>5:
        break
    ct+=1

(u'225522343', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 

In [ ]:
for m in common_event:
    if common_event[m]==15776612:
        print(m)
        break

pcnlrlyvlbmc


In [ ]:
len(common1_event)

115571

In [ ]:
if 1 in common1_event[u'wljghlytmbzb']:
    print(1)
else: print(0)

1


## Taking mem_at file and making set for search member files

In [ ]:

#It gives the set of (event-id, event_time) for all the events attended by each user.
ct=0
mem_at=pickle.load(open('mem_at','r'))
print(len(mem_at))
for m in mem_at:
    print("NEW ID  ",m,mem_at[m])
    if ct>5:
        break
    ct+=1

223788
('NEW ID  ', 57671682, set([(u'74913292', 1344724200000L)]))
('NEW ID  ', 188219397, set([(u'227073685', 1450224000000L)]))
('NEW ID  ', 100663302, set([(u'dhzjmgyrnbnc', 1383175800000L), (u'dhzjmgyrlblc', 1377732600000L), (u'dhzjmgyrnbdb', 1380756600000L), (u'dhzjmgyrqbxb', 1387413000000L), (u'dhzjmgyrlbcc', 1377127800000L), (u'dhzjmgyrnbvb', 1381966200000L), (u'dhzjmgyrlbsb', 1376523000000L)]))
('NEW ID  ', 90701832, set([(u'166281682', 1400626800000L)]))
('NEW ID  ', 200802315, set([(u'ghsfrlyvfbnc', 1459381500000L)]))
('NEW ID  ', 152043532, set([(u'xvgznyskbdb', 1404345600000L)]))
('NEW ID  ', 180180312, set([(u'211533242', 1414884600000L)]))


In [ ]:
mem_at_listValue={}
for m in mem_at:
    mem_at_listValue[m]=list(mem_at[m])

In [ ]:
ct=0
for m in mem_at_listValue:
    print(m,mem_at_listValue[m])
    if ct>5:
        break
    ct+=1

(57671682, [(u'74913292', 1344724200000L)])
(188219397, [(u'227073685', 1450224000000L)])
(100663302, [(u'dhzjmgyrnbnc', 1383175800000L), (u'dhzjmgyrlblc', 1377732600000L), (u'dhzjmgyrnbdb', 1380756600000L), (u'dhzjmgyrqbxb', 1387413000000L), (u'dhzjmgyrlbcc', 1377127800000L), (u'dhzjmgyrnbvb', 1381966200000L), (u'dhzjmgyrlbsb', 1376523000000L)])
(90701832, [(u'166281682', 1400626800000L)])
(89874202, [(u'115185542', 1368486000000L), (u'116748212', 1370698200000L), (u'133377602', 1377451800000L), (u'231940879', 1467154800000L), (u'232615751', 1470265200000L), (u'233040422', 1471993200000L), (u'135661172', 1377729000000L), (u'232895599', 1470351600000L), (u'123906392', 1373497200000L), (u'232454866', 1469142000000L), (u'233092241', 1470783600000L), (u'232475656', 1469055600000L), (u'232224892', 1469138400000L), (u'232281431', 1469658600000L), (u'115794812', 1370473200000L), (u'232876160', 1470178800000L)])
(200802315, [(u'ghsfrlyvfbnc', 1459381500000L)])
(152043532, [(u'xvgznyskbdb', 14

In [ ]:
mem_at_listValue[57671682][0][0]

u'74913292'

In [ ]:
#making list and set of keys of mtags file
mtags_keys=[]
for m in mtags:
    #print(m,mtags[m])
    mtags_keys.append(m)
print(len(mtags_keys))
print(mtags_keys[0:5])
set_mtags_keys=set(mtags_keys)

423763
[57671682, 100663302, 209715208, 127926282, 152043532]


In [ ]:
##making list and set of keys of mem_at file
mem_at_keys=[]
for m in mem_at:
    #print(m,mtags[m])
    mem_at_keys.append(m)
print(len(mem_at_keys))
print(mem_at_keys[0:5])
set_mem_at_keys=set(mem_at_keys)

223788
[57671682, 188219397, 100663302, 90701832, 200802315]


In [ ]:
#taking common of two sets
intersection=(list(set_mtags_keys.intersection(set_mem_at_keys)))
len(intersection)

223393

In [ ]:
##creating new mem+at file for common mtags and mem_at files
new_mem_at={}
for m in mem_at_listValue:
    if m in intersection:
        new_mem_at[m]=mem_at_listValue[m]

KeyboardInterrupt: 

In [ ]:
ct=0
for m in new_mem_at:
    print(m,new_mem_at[m])
    if ct>5:
        break
    ct+=1
len(new_mem_at)

In [ ]:
##Here I create list of member id's whose events are in common
listOfEvent=[]
for i in new_mem_at:
    for j in range(len(new_mem_at[i])):
        if new_mem_at[i][j][0] in common:
            listOfEvent.append(i)
            break

In [ ]:
print(listOfEvent[:5])
len(listOfEvent)

[100663302, 90701832, 89874202, 115343382, 160432162]


125927

In [ ]:
##Here i make a dictionary of mtags i.e., member id and mtags
mtagsCommon={}
for i in listOfEvent:
    mtagsCommon[i]=list(mtags[i])

In [ ]:
##Here i make a dictionary of mtags i.e., member id and events from mem at file
mem_atCommon={}
for i in listOfEvent:
    mem_atCommon[i]=new_mem_at[i]

In [ ]:
ct=0
print(len(mem_atCommon))
for m in mem_atCommon:
    print(m,mem_atCommon[m])
    if ct>2:
        break
    ct+=1

125927
(100663302, [(u'dhzjmgyrnbnc', 1383175800000L), (u'dhzjmgyrlblc', 1377732600000L), (u'dhzjmgyrnbdb', 1380756600000L), (u'dhzjmgyrqbxb', 1387413000000L), (u'dhzjmgyrlbcc', 1377127800000L), (u'dhzjmgyrnbvb', 1381966200000L), (u'dhzjmgyrlbsb', 1376523000000L)])
(90701832, [(u'166281682', 1400626800000L)])
(12320779, [(u'230288760', 1461709800000L), (u'drmwwhysqbvb', 1418778000000L), (u'pdbfdlyvjblc', 1467154800000L), (u'227888680', 1462919400000L), (u'234195139', 1476138600000L), (u'230275055', 1462125600000L), (u'230783910', 1462645800000L), (u'234531526', 1477749600000L)])
(8650765, [(u'226374145', 1449435600000L), (u'15092527', 1288468800000L), (u'85501952', 1350849600000L), (u'9589377', 1234040400000L), (u'85212832', 1350244800000L), (u'37813182', 1319410800000L), (u'37636542', 1319407200000L)])


## Here I create the acjacency list

In [ ]:
##Here I take only member ID and events and drop the time format
mem_event_final={}
for i in mem_atCommon:
    mem_event_final[i]=[x[0] for x in mem_atCommon[i]]

In [ ]:
ct=0
print(len(mem_event_final))
for m in mem_atCommon:
    print(m,mem_event_final[m])
    if ct>2:
        break
    ct+=1

125927
(100663302, [u'dhzjmgyrnbnc', u'dhzjmgyrlblc', u'dhzjmgyrnbdb', u'dhzjmgyrqbxb', u'dhzjmgyrlbcc', u'dhzjmgyrnbvb', u'dhzjmgyrlbsb'])
(90701832, [u'166281682'])
(12320779, [u'230288760', u'drmwwhysqbvb', u'pdbfdlyvjblc', u'227888680', u'234195139', u'230275055', u'230783910', u'234531526'])
(8650765, [u'226374145', u'15092527', u'85501952', u'9589377', u'85212832', u'37813182', u'37636542'])


In [ ]:
ct=0
print(len(mtagsCommon))
for m in mtagsCommon:
    print(m,mtagsCommon[m])
    if ct>5:
        break
    ct+=1

125927
(100663302, [])
(90701832, [u'Business Intelligence', u'Technology', u'Big Data'])
(12320779, [u'Dining Out', u'Expat', u'Foodie', u'Hiking', u'Travel', u'Cooking', u'International Travel', u'Taiwan International Meetup', u'Cooking Dinner Parties', u'Culture Exchange', u'New In Town', u'Language Exchange', u'Adventure', u'Social', u'Recipes', u'Mandarin Learning'])
(8650765, [])
(12845070, [])
(115343382, [])
(17563671, [])


In [ ]:
#Here take the values of above dictionary
mtagValues=[]
for m in mtagsCommon:
    mtagValues.append(list(mtagsCommon[m]))
print(len(mtagValues))
print((mtagValues[0]))
# for i in range(5):
#     print((mtagValues[i]))
#     print(len(mtagValues[i]))

125927
[]


In [ ]:
#print((mtagValues[1]))
for i in range(5):
    print((mtagValues[i]))
    print(len(mtagValues[i]))

[]
0
[u'Business Intelligence', u'Technology', u'Big Data']
3
[u'Dining Out', u'Expat', u'Foodie', u'Hiking', u'Travel', u'Cooking', u'International Travel', u'Taiwan International Meetup', u'Cooking Dinner Parties', u'Culture Exchange', u'New In Town', u'Language Exchange', u'Adventure', u'Social', u'Recipes', u'Mandarin Learning']
16
[]
0
[]
0


In [ ]:
listOfmtags=[]
#m=[]
for i in range(len(mtagValues)):
    #m.append(intersection[i])

    for j in range(len(mtagValues[i])):

        listOfmtags.append(mtagValues[i][j])
print("len of listOfmtags= ",len(listOfmtags))
#print("len of m= ",len(m))
listOfmtags[0:8]
l1=list(set(listOfmtags))
print(len(set(listOfmtags)))
#m[0:5]

('len of listOfmtags= ', 1705500)
24072


In [ ]:
from collections import Counter
hset=Counter(listOfmtags);
mostOccure=hset.most_common(1000)
(mostOccure[0:10])

[(u'Dining Out', 26475),
 (u'Fitness', 25318),
 (u'Live Music', 24399),
 (u'Outdoors', 23717),
 (u'Travel', 21631),
 (u'Art', 17871),
 (u'Hiking', 16488),
 (u'Watching Movies', 15591),
 (u'Chicago', 15030),
 (u'Startup Businesses', 14691)]

In [ ]:
#Taking the most common mtags in list
mostOccuremtags=[]
for i in range(1000):
    for j in range(1):
        mostOccuremtags.append(mostOccure[i][j])
mostOccuremtags[0:5]

[u'Dining Out', u'Fitness', u'Live Music', u'Outdoors', u'Travel']

In [ ]:
len(mostOccuremtags)

1000

In [ ]:
##creating dictionary of intersection elements as key and value is zeroes
valueZero={}
for i in mtagsCommon:
    valueZero[i]=np.zeros((1, len(mostOccuremtags)+1))

In [ ]:
len(valueZero[100663302][0])

1001

## Here I get the mtags dicts final of 125927*1001 matrix type

In [ ]:
for i in valueZero:
    if (len(mtagsCommon[i]))==0:
        valueZero[i][0][1000]=1
    else:
        for j in range(len(mtagsCommon[i])):
            for k in range(1000):
                if mtagsCommon[i][j]==mostOccuremtags[k]:
                    valueZero[i][0][k]=1


#     for j in range(len(mtagsCommon[i])):
#         if (len(mtagValues[i]))==0:
#             valueZero[i][0][1000]=1

In [ ]:
for i in valueZero:
    valueZero[i]=valueZero[i].tolist()

In [ ]:
for i in valueZero:
    valueZero[i]=valueZero[i][0]

In [ ]:
ct=0
for m in valueZero:
    print(m,valueZero[m])
    if ct>5:
        break
    ct+=1

(41680902, [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0

In [ ]:
len(list(valueZero.keys())),len(list(mtagsCommon.keys()))

(125927, 125927)

In [ ]:
m

## creating event duration vectors from common events

In [ ]:
## Here I create a dict of common event and event place id
common_eventDuration={}
for i in common:
    common_eventDuration[i]=e_duration[i]

In [ ]:
len(common_eventDuration)

115571

In [ ]:
#making e_time of common events
new_et={}
for m in common:
    new_et[m]=int(et[m])
len(new_et)

115571

In [ ]:
import datetime
for m in new_et:
    s= (new_et[m]) / 1000.0
    new_et[m]=(datetime.datetime.fromtimestamp(s).strftime('%Y-%m-%d %H:%M:%S.%f'))

In [ ]:
ct=0
for m in new_et:
    print(m,new_et[m])
    if ct>10:
        break
    ct+=1

(u'qwbvpkysmbdc', '2014-09-23 05:00:00.000000')
(u'225522343', '2015-09-27 05:30:00.000000')
(u'170374282', '2014-06-21 21:30:00.000000')
(u'dvcrlgyrnbqb', '2013-10-12 20:00:00.000000')
(u'thqsclysqbvb', '2014-12-17 05:30:00.000000')
(u'dwldlgysfbzb', '2014-03-20 05:00:00.000000')
(u'dtlcxjytcbfb', '2015-01-04 00:30:00.000000')
(u'qlbbhhysgbcb', '2014-04-02 05:30:00.000000')
(u'192062632', '2014-07-28 00:30:00.000000')
(u'qmpxfgysdbnb', '2014-02-11 06:30:00.000000')
(u'dwldlgytjbnb', '2015-06-11 05:00:00.000000')
(u'dlzvbfyrcbtb', '2013-01-15 23:30:00.000000')


In [ ]:
from datetime import datetime

# Convert each date in the data to a datetime object and sort based on the date
sorted_data = sorted(new_et.items(), key=lambda x: datetime.strptime(x[1], '%Y-%m-%d %H:%M:%S.%f'))

# Convert the sorted tuples back to a dictionary
sorted_dict = dict(sorted_data)

#print(sorted_dict)


In [ ]:
type(sorted_data)

list

In [ ]:
sorted_data=[sorted_data[i][0] for i in range(len(sorted_data))]

In [ ]:
if sorted(sorted_data)==sorted(list(new_et.keys())):
    print(1)

1


In [ ]:
ct=0
for m in sorted_data:
    print(m)
    if ct>10:
        break
    ct+=1

61700742
qcwdpcyqhbqb
dmjgwcyqhbqb
62051942
60865242
64377012
63862642
43257942
64337002
62475382
64379022
61196172


In [ ]:
##Here I save the list to json with maintaining the order
import json
from collections import OrderedDict

# Sample list
#my_list = ['apple', 'banana', 'orange', 'grape']

# Convert list to an ordered dictionary to maintain order
ordered_dict = OrderedDict((str(i), item) for i, item in enumerate(sorted_data))

# Write the ordered dictionary to a JSON file while preserving order
with open('ordered_events.json', 'w') as f:
    json.dump(ordered_dict, f, indent=2)

In [ ]:
from datetime import datetime

data = {
    (u'qwbvpkysmbdc', '2014-09-23 05:00:00.000000'),
    (u'225522343', '2015-09-27 05:30:00.000000'),
    (u'170374282', '2014-06-21 21:30:00.000000'),
    (u'dvcrlgyrnbqb', '2013-10-12 20:00:00.000000'),
    (u'thqsclysqbvb', '2014-12-17 05:30:00.000000'),
    (u'dwldlgysfbzb', '2014-03-20 05:00:00.000000'),
    (u'dtlcxjytcbfb', '2015-01-04 00:30:00.000000')
}

# Convert each date in the data to a datetime object and sort based on the date
sorted_data = sorted(data, key=lambda x: datetime.strptime(x[1], '%Y-%m-%d %H:%M:%S.%f'))
sorted_dict = {item[0]: item[1] for item in sorted_data}
print(sorted_data)

[(u'dvcrlgyrnbqb', '2013-10-12 20:00:00.000000'), (u'dwldlgysfbzb', '2014-03-20 05:00:00.000000'), (u'170374282', '2014-06-21 21:30:00.000000'), (u'qwbvpkysmbdc', '2014-09-23 05:00:00.000000'), (u'thqsclysqbvb', '2014-12-17 05:30:00.000000'), (u'dtlcxjytcbfb', '2015-01-04 00:30:00.000000'), (u'225522343', '2015-09-27 05:30:00.000000')]


In [ ]:
ct=0
for m in common_eventDuration:
    print(m,common_eventDuration[m])
    if ct>5:
        break
    ct+=1

(u'qwbvpkysmbdc', 1.0)
(u'225522343', 6.0)
(u'170374282', 6.0)
(u'dvcrlgyrnbqb', 1.0)
(u'thqsclysqbvb', 1.0)
(u'dwldlgysfbzb', 2.0)
(u'dtlcxjytcbfb', 6.0)


In [ ]:
if list(new_et.keys())==list(common_eventDuration.keys()):
    print(1)

1


In [ ]:
k={}
for m in new_et:
    k[m]=[0]*48

In [ ]:
from datetime import datetime
for m in k:
    date_string = new_et[m]
    date_format = '%Y-%m-%d %H:%M:%S.%f'
    datetime_obj = datetime.strptime(date_string, date_format)
    weekday = datetime_obj.weekday()
    hour = int(datetime_obj.strftime('%H'))

    if weekday < 5:
        if int(common_eventDuration[m])==0:
            k[m][hour]=1
        elif 24<int(common_eventDuration[m])<=120:
            for i in range(24):
                k[m][i]=1
        elif int(common_eventDuration[m])>120:
            for i in range(48):
                k[m][i]=1

        else:
            for i in range(int(common_eventDuration[m])+1):
                if hour<24:
                    k[m][hour]=1
                    hour+=1
                else:
                    #print(m)
                    #print(hour-24)
                    k[m][hour-24]=1
                    hour+=1


    else:  # 5 Sat, 6 Sun
        if int(common_eventDuration[m])==0:
            k[m][hour]=1
        elif 24<int(common_eventDuration[m])<=48:
            for i in range(24):
                k[m][i]=1
        elif int(common_eventDuration[m])>48:
            for i in range(48):
                k[m][i]=1
        else:
            for i in range(int(common_eventDuration[m])+1):
                if hour<24:
                    k[m][hour+24]=1
                    hour+=1
                else:
                    k[m][48-hour]=1
                    hour+=1

In [ ]:
ct=0
for m in common_eventDuration:
    #print(m,new_e_duration[m])
    if common_eventDuration[m]>120:
        print(m)
        break
for m in common_eventDuration:
    #print(m,new_e_duration[m])
    if common_eventDuration[m]<24:
        print(m)
        break

220669348
qwbvpkysmbdc


In [ ]:
print("the time is",new_et[u'qwbvpkysmbdc'])
print("duration is", common_eventDuration[u'qwbvpkysmbdc'])

print(k[u'qwbvpkysmbdc'])
date_string = new_et[u'qwbvpkysmbdc']
date_format = '%Y-%m-%d %H:%M:%S.%f'
datetime_obj = datetime.strptime(date_string, date_format)
weekday = datetime_obj.weekday()
print("day of the week",weekday)

('the time is', '2014-09-23 05:00:00.000000')
('duration is', 1.0)
[0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
('day of the week', 1)


## Here we get time vectors

In [ ]:
ct=0
for m in k:
    print(m,k[m])
    if ct>5:
        break
    ct+=1

(u'225522343', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
(u'170374282', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1])
(u'dvcrlgyrnbqb', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0])
(u'thqsclysqbvb', [0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
(u'dwldlgysfbzb', [0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
(u'dtlcxjytcbfb', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
(u'qlbbhhysgbcb', [0, 

In [ ]:
len(k)

115571

## Now we will find the event tags vector

In [ ]:
event_tags={}
#ename_CH=pickle.load(open('ename_bhnoda_155_CH','r'))
#print(len(ename_CH))
for i in range(len(sorted_data)):
#for m in sorted_data:
    event_tags[sorted_data[i]]=ename_CH[sorted_data[i]]

len(event_tags)

115571

In [ ]:
#my_dict = {(element):ename_CH[element]  for element in (sorted_data)}

In [ ]:
ename_CH[u'61700742']
list(event_tags.keys())[:5]

[u'pgtkhlytmbfc',
 u'qztqdhysjbwb',
 u'dvcrlgyrnbqb',
 u'thqsclysqbvb',
 u'dwldlgysfbzb']

In [ ]:
ct=0
print(sorted_data[:5])
for m in event_tags:
    print(m,event_tags[m])
    if ct>2:
        break
    ct+=1

[u'61700742', u'qcwdpcyqhbqb', u'dmjgwcyqhbqb', u'62051942', u'60865242']
(u'pgtkhlytmbfc', u"Let's meet and get our groove on,  maybe sweat a little as we go....")
(u'qztqdhysjbwb', u'Pathfinder Adventure Path at The Gaming Goat: Skull Shackles')
(u'dvcrlgyrnbqb', u"Beginner's Meditation Program")
(u'thqsclysqbvb', u'Walk & Talk! (INDOORS).  NEW MEMBER INVITATION NIGHT!')


In [ ]:
from bs4 import BeautifulSoup
import re
from rake_nltk import Rake
r = Rake()
ll=[]

def remove_links(html):
    soup = BeautifulSoup(html, 'html.parser')
    # Remove anchor tags
    for a in soup.find_all('a'):
        a.extract()
    # Remove any remaining link-like text
    clean_text = re.sub(r'\b\w+://\S+~', '', soup.get_text())
    return clean_text
def remove_pattern(text, pattern):
    cleaned_text = text.replace(pattern, '.')
    return cleaned_text
def remove_lines_with_pattern(text):
    text=re.sub('https?://\S+|www\.\S+','',text)
    #text=re.sub('[%s]'%re.escape(string.punctuation),'',text)
#     text=[word for word in text.split(' ') if word not in stopword]
#     text=" ".join(text)
    return text
for m in event_tags:
    html_paragraph = event_tags[m]
    #pattern = r'Purchase Class: <a href=".+?">'
    clean_paragraph_0 = remove_links(html_paragraph)
    cleaned_paragraph_1 = remove_pattern(clean_paragraph_0, '~~~')
    cleaned_paragraph = remove_lines_with_pattern(cleaned_paragraph_1)
    text =cleaned_paragraph
    r.extract_keywords_from_text(text)
    ranked_phrases = r.get_ranked_phrases()[:2]
    event_tags[m]=ranked_phrases
    #print(ranked_phrases)
    #break
    for phrase in ranked_phrases:
        ll.append(phrase)


#print(cleaned_paragraph)
#print(cleaned_paragraph_1)

C:\Users\hp\anaconda3\envs\py27\lib\site-packages\bs4\builder\_htmlparser.py:78: UserWarning: EOF in middle of entity or char ref
  warnings.warn(msg)
C:\Users\hp\anaconda3\envs\py27\lib\site-packages\bs4\__init__.py:294: UserWarning: "." looks like a filename, not markup. You should probably open this file and pass the filehandle into Beautiful Soup.
  ' Beautiful Soup.' % markup)


In [ ]:
len(ll)

201120

In [ ]:
from collections import Counter
hset=Counter(ll);
ll=hset.most_common(500)

In [ ]:
(ll[:25])

[(u'smart recovery meeting', 1044),
 (u'chicago', 941),
 (u'yoga', 887),
 (u'introduction', 815),
 (u'speed dating', 654),
 (u'single professionals', 647),
 (u'meditation', 630),
 (u'tai chi', 539),
 (u'let', 532),
 (u'energy healing martial arts', 507),
 (u'lincoln park', 506),
 (u'evanston', 472),
 (u'beginners', 461),
 (u'meet', 452),
 (u'bowing meditation', 452),
 (u'karate', 429),
 (u'morning', 425),
 (u'salsa class', 402),
 (u'singles w', 391),
 (u'jka karate', 388),
 (u'aikido introduction', 382),
 (u'community acupuncture', 362),
 (u'weight loss', 360),
 (u'meditate', 360),
 (u'wicker park', 343)]

In [ ]:
##Here I creating the list of 500 event tags
listOfTags=[]
for i in range(len(ll)):
    listOfTags.append(ll[i][0])

In [ ]:
import copy
#new_list = copy.copy(old_list)
l3=copy.copy(listOfTags)
l3

[u'smart recovery meeting',
 u'chicago',
 u'yoga',
 u'introduction',
 u'speed dating',
 u'single professionals',
 u'meditation',
 u'tai chi',
 u'let',
 u'energy healing martial arts',
 u'lincoln park',
 u'evanston',
 u'beginners',
 u'meet',
 u'bowing meditation',
 u'karate',
 u'morning',
 u'salsa class',
 u'singles w',
 u'jka karate',
 u'aikido introduction',
 u'community acupuncture',
 u'weight loss',
 u'meditate',
 u'wicker park',
 u'singing bowl meditation',
 u'women',
 u'pick',
 u'join us',
 u'fusion yoga',
 u'svetlana raoufi',
 u'oak park',
 u'word',
 u'lakeview',
 u'kundalini yoga',
 u'study god',
 u'meetup',
 u'real estate investors meeting',
 u'healing clinic',
 u'open gaming',
 u'open class',
 u'beginner',
 u'play',
 u'stop chasing customers',
 u'mantra meditation',
 u'open gym',
 u'networking',
 u'schaumburg',
 u'welcome',
 u'free meditation',
 u'network',
 u'4 week series',
 u'indoor target archery',
 u'oneness blessing',
 u'workshop',
 u'work',
 u'r',
 u'free class',
 u'wic

In [ ]:
len(l3)

500

In [ ]:
ct=0
for m in event_tags:
    print(m,event_tags[m])
    if ct>7:
        break
    ct+=1

(u'pgtkhlytmbfc', u"Let's meet and get our groove on,  maybe sweat a little as we go....")
(u'qztqdhysjbwb', u'Pathfinder Adventure Path at The Gaming Goat: Skull Shackles')
(u'dvcrlgyrnbqb', u"Beginner's Meditation Program")
(u'thqsclysqbvb', u'Walk & Talk! (INDOORS).  NEW MEMBER INVITATION NIGHT!')
(u'dwldlgysfbzb', u'Invite in Peace and Happiness...')
(u'dtlcxjytcbfb', u'Saturday Open Table Top Gaming')
(u'qlbbhhysgbcb', u'Kundalini Yoga & Meditation')
(u'192062632', u'Manga Jam #9: Sexy Voice and Robo')
(u'qmpxfgysdbnb', u'Shorinji Kempo: Introduction')


In [ ]:
sorted_data[:5]

[u'61700742', u'qcwdpcyqhbqb', u'dmjgwcyqhbqb', u'62051942', u'60865242']

In [ ]:
import copy
ct=0
event_tagsFinal={}
for m in event_tags:
    l3=copy.copy(listOfTags)
    for i in range(len(listOfTags)):
        if l3[i] in event_tags[m]:
            l3[i]=1
        else:
            l3[i]=0

    event_tagsFinal[m]=l3
#     ct+=1
#     if ct>5:
#         break

## Here we get event tags arrays

In [ ]:
ct=0
for m in event_tags:
    print(m,event_tagsFinal[m])
    if ct>7:
        break
    ct+=1

(u'pgtkhlytmbfc', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 

In [ ]:
for i in event_tags:
    if event_tags[i][0] or event_tags[i][1] in ll:
        print(i)
        break


pgtkhlytmbfc


In [ ]:
print(event_tags[u'pgtkhlytmbfc'])

Let's meet and get our groove on,  maybe sweat a little as we go....


In [ ]:
if 1 in event_tagsFinal[u'pgtkhlytmbfc']:
    print("1")
else:
    print("0")

1


In [ ]:
len(event_tagsFinal)

115571

## start the event vector final dictionary

In [ ]:
marged_dic={}
for i in event_tags:
    marged_dic[i]=event_tagsFinal[i]+k[i]+common1_event[i]

In [ ]:
len(marged_dic[u'170374282'])

1548

In [ ]:
from collections import OrderedDict
ordered_dict = OrderedDict((key, marged_dic[key]) for key in sorted_data if key in marged_dic)

In [ ]:
ct=0
for m in ordered_dict:
    print(m,ordered_dict[m])
    break
    if ct>2:
        break
    ct+=1

(u'61700742', [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0

In [ ]:
type(ordered_dict)

collections.OrderedDict

In [ ]:
import json

In [ ]:
# Specify the file path where you want to save the JSON data
file_path = 'merged_data.json'

In [ ]:
##create json file for event
with open(file_path, 'w') as json_file:
    json.dump(ordered_dict, json_file)

print("Data saved to JSON file:", file_path)

('Data saved to JSON file:', 'merged_data.json')


## create json file for members

In [ ]:
# Specify the file path where you want to save the JSON data
file_path = 'mem_merged_data.json'##create json file for event
with open(file_path, 'w') as json_file:
    json.dump(valueZero, json_file)

print("Data saved to JSON file:", file_path)

('Data saved to JSON file:', 'mem_merged_data.json')


In [ ]:

# Specify the file path where you want to save the JSON data
file_path = 'mem_event_data.json'##create json file for event
with open(file_path, 'w') as json_file:
    json.dump(mem_event_final, json_file)

print("Data saved to JSON file:", file_path)

('Data saved to JSON file:', 'mem_event_data.json')
